### PR TITLE
added catch when using trusted auth since no user is logged in

### PR DESCRIPTION
**Pull request by johnloyimqs**

### CODE REVIEW

File: imqsauth/http.go [hunk 1]
```diff
@@ -721,7 +721,10 @@ func httpHandlerCreateUser(central *ImqsCentral, w http.ResponseWriter, r *httpR
 	}
 
 	// Get the userId of the logged-in user
-	createdby := r.token.UserId
+	var createdby authaus.UserId
+	if !strings.HasPrefix(r.http.Header.Get("Authorization"), "IMQS-Trusted ") {
```
Comment (bmharper) [inline imqsauth/http.go]: I think this would be better:
```go
if r.token != nil {
    createdby = r.token.UserId
}
```

Comment (bmharper) [inline imqsauth/http.go]: I think this would be better:
```go
if r.token != nil {
    createdby = r.token.UserId
}
```

Comment (johnloyimqs) [reply imqsauth/http.go]: OK, agree. The called would have had to be authenticated with admin privileges (handlerFlagNeedAdminRights) higher up in the code, so updated.

Comment (johnloyimqs) [reply imqsauth/http.go]: OK, agree. The called would have had to be authenticated with admin privileges (handlerFlagNeedAdminRights) higher up in the code, so updated.

**Pull request merged**